### PR TITLE
Various fix for MacOS X 10.10, Linux and zsh

### PR DESCRIPTION
**Pull request by odtvince**
Fix DNS cache flushing for MacOS X 10.10 to 10.10.3
Fix date and sw_vers for Linux
Fix escaping for zsh

### CODE REVIEW

File: adblock [hunk 1]
```diff
@@ -10,12 +10,17 @@ ORIGINAL='/etc/hosts.without-adblock'
...
 clear_dns_cache() {
-	if sw_vers -productVersion | grep -q 10.6
-	then sudo dscacheutil -flushcache
-	else sudo killall -HUP mDNSResponder
-	fi
+	if command -v sw_vers >/dev/null 2>&1
+	then
+		if sw_vers -productVersion | grep -q '10\.6'
+		then sudo dscacheutil -flushcache
+		elif sw_vers -productVersion | grep -q '10\.10[\.123]*'
```
Comment (MattiSG) [inline adblock]: 10.10.4 seems to be using `mDNSResponder`, as before (at least I can `pgrep` it while I can't `discoveryd`), yet your regexp matches `10.10.4` due to the trailing `*`. Are you sure you meant the `*`?

**Comment by MattiSG** [comment]
Thanks, I think this fixes #1!  :)  On which OS did you test this?

**Comment by odtvince** [comment]
Tested on Debian Jessie and MacOS X 10.10.3

File: adblock (hunk 1 above)
Comment (odtvince) [inline adblock]: No, you're right. It should be ''?$'' instead.

**Comment by MattiSG** [comment]
Thanks @odtvince!  :)

**Pull request merged**